ford/sourceform.py

```diff
@@ -1316,12 +1324,11 @@ def strip_type(s):
             and isinstance(self, FortranType)
             and hasattr(self, "extends")
         ):
-            extend = self
-            while getattr(extend, "extends", None) is not None:
-                if extend.extends.name.lower() == c:
-                    call_type = extend.extends
+            extend: Optional[FortranCodeUnit] = self
+            while extend := getattr(extend, "extends", None):
+                if extend.name.lower() == call.name:
```

Comment:
@JosephWood2001 Please could you take a quick look at this function? I noticed that here `c` wasn't defined, I think it should've been `call.name`, but I wasn't able to quickly come up with a test case that hits this